app/src/main/java/com/example/chooseyouowncocktail_g2/adapter/DrinkCardAdapter.kt

```diff
@@ -0,0 +1,49 @@
+package com.example.chooseyouowncocktail_g2.adapter
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.example.chooseyouowncocktail_g2.model.Beer
+import com.example.chooseyouowncocktail_g2.DrinkList
+import com.example.chooseyouowncocktail_g2.databinding.DrinkCardBinding
+
+class DrinkCardAdapter(
+    var languageList: List<Beer>,
+) : RecyclerView.Adapter<DrinkCardAdapter.ViewHolder>() {
+
+    // create an inner class with name ViewHolder
+    // It takes a view argument, in which pass the generated class of single_item.xml
+    // ie SingleItemBinding and in the RecyclerView.ViewHolder(binding.root) pass it like this
+    inner class ViewHolder(val binding: DrinkCardBinding) : RecyclerView.ViewHolder(binding.root)
+
+    // inside the onCreateViewHolder inflate the view of SingleItemBinding
+    // and return new ViewHolder object containing this layout
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+        val binding = DrinkCardBinding.inflate(LayoutInflater.from(parent.context), parent, false)
+
+        return ViewHolder(binding)
+    }
+
+    // bind the items with each item
+    // of the list languageList
+    // which than will be
+    // shown in recycler view
+    // to keep it simple we are
+    // not setting any image data to view
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        with(receiver = holder) {
+            with(DrinkList.beerList()[position]) {
```

Comment:
Se hai la lista nel costruttore dell'adapter non serve prenderla da altre parti

Reply:
Non ho modificato il codice e il nome della variabile perchè volevo prima capire cosa avesse fatto Michele
In ogni caso ha detto che sistemerà lui stesso